modules/nextflow/src/test/groovy/nextflow/k8s/model/PodSpecBuilderTest.groovy

```diff
@@ -785,6 +785,27 @@ class PodSpecBuilderTest extends Specification {
         'x2345678901234567890123456789012345678901234567890123456789012345' | 'x23456789012345678901234567890123456789012345678901234567890123'
     }
 
+    @Unroll
+    def 'should sanitize k8s label key: #label_key' () {
+        given:
+        def builder = new PodSpecBuilder()
+
+        expect:
+        builder.sanitizeKey(label_key, PodSpecBuilder.MetaType.LABEL) == str
+
+        where:
+        label_key               | str
+        'foo'                   | 'foo'
+        'key 1'                 | 'key_1'
+        'foo.bar/key 2'         | 'foo.bar/key_2'
+        'foo.bar/key 2/key 3'   | 'foo.bar/key_2_key_3'
+        'foo.bar/'              | 'foo.bar'
+        '/foo.bar'              | 'foo.bar'
+        'x2345678901234567890123456789012345678901234567890123456789012345' | 'x23456789012345678901234567890123456789012345678901234567890123'
```

Comment:
Since `/` has a special syntax, not sure anymore it should be allowed to provide two or more or them or to allow leading or trailing slashes. Would not be better to report an error?

Reply:
The reason why I didn't trigger any error is because I think it is consistent behavior with the current one, sort of "I will fix it for you" philosophy, i.e. if you input invalid characters they are just replaced with '_' instead of reporting an error. 

But I can switch it to throw an `IllegalArgumentException` if there are more than on `/` if you think is best. 

What do you think about leading and trailing slashes? Would yo prefer an error or sanitizing as it is now? Currently in this case I remove the slash and consider it a key without prefix.

Reply:
My understanding is that `/` as special semantic in K8s labels and annotation, therefore if it's wrong formatted i.e. two or more `/` or leading/trailing slashes is very likely the user is messing up thing. Therefore it would be better to throw an error? do you agree? if so I can do quickly

Reply:
I agree @pditommaso. I think is better to get an error so the user can just adjust the parameters. 

I just pushed the fix and the tests for it.